### PR TITLE
Clear the file upload input after success call

### PR DESCRIPTION
**Pull request by YaiyaSam**
## Purpose

The purpose of this PR is to make changes requested under pr #63

## Goals

* To clear out the input field after successfully uploading a file. 
* To make changes to the code to maintain the consistency with the rest of the codebase. 

## Comments
* I have left the `react-toastify` library for now.

### CODE REVIEW

File: admin-portal/src/pages/needUpload/needUpload.tsx [hunk 1]
```diff
@@ -1,59 +1,55 @@
...
+          position: "top-right",
+          autoClose: 5000,
+          hideProgressBar: true,
+          closeOnClick: true,
+          pauseOnHover: true,
+          draggable: true,
+          progress: undefined,
         });
+        setFileName("");
+      } catch (e: any) {
```
Comment (jayasanka-sack) [inline admin-portal/src/pages/needUpload/needUpload.tsx:37]: Don't use `any`.

File: admin-portal/src/pages/needUpload/needUpload.tsx [hunk 2]
```diff
@@ -1,59 +1,55 @@
...
+      try {
+        const response = await AidPackageService.postNeeds(formData);
+        toast.success(response.data, {
+          position: "top-right",
+          autoClose: 5000,
+          hideProgressBar: true,
+          closeOnClick: true,
+          pauseOnHover: true,
+          draggable: true,
+          progress: undefined,
```
Comment (jayasanka-sack) [inline admin-portal/src/pages/needUpload/needUpload.tsx:34]:
```suggestion
```

File: admin-portal/src/pages/needUpload/needUpload.tsx [hunk 3]
```diff
@@ -1,59 +1,55 @@
...
-import "react-toastify/dist/ReactToastify.css";
+import { toast } from "react-toastify";
 
 export function NeedUpload() {
   const [file, setFile] = useState(null);
+  const [fileName, setFileName] = useState("");
   const [errorList, setErrorList] = useState([]);
-  const navigate = useNavigate();
 
   function handleChange(event: any) {
```
Comment (jayasanka-sack) [inline admin-portal/src/pages/needUpload/needUpload.tsx:13]: Don't use `any`

```suggestion
  function handleChange(event: ChangeEvent<HTMLInputElement>) {
```

File: admin-portal/src/pages/needUpload/needUpload.tsx [hunk 4]
```diff
@@ -1,59 +1,55 @@
...
     setErrorList([]);
     setFile(event.target.files[0]);
+    setFileName(event.target.value);
   }
 
   const handleSubmit = async (event: any) => {
```
Comment (jayasanka-sack) [inline admin-portal/src/pages/needUpload/needUpload.tsx:19]: any

File: admin-portal/src/pages/needUpload/needUpload.tsx [hunk 5]
```diff
@@ -1,59 +1,55 @@
...
     event.preventDefault();
-
     const formData = new FormData();
 
     if (file != undefined) {
```
Comment (jayasanka-sack) [inline admin-portal/src/pages/needUpload/needUpload.tsx:23]:
```suggestion
    if (file) {
```

**Review by jayasanka-sack — APPROVED** [review]
Thanks for the PR, Kalsha.
I added some comments

File: admin-portal/src/pages/needUpload/needUpload.tsx (hunk 1 above)
Comment (YaiyaSam) [reply admin-portal/src/pages/needUpload/needUpload.tsx:37]: what can i use here instead?

Comment (YaiyaSam) [reply admin-portal/src/pages/needUpload/needUpload.tsx:37]: It says
> Catch clause variable type annotation must be 'any' or 'unknown' if specified.

Comment (jayasanka-sack) [reply admin-portal/src/pages/needUpload/needUpload.tsx:37]: It should be: 

```
 } catch (error) {
      if (error instanceof AxiosError) {
```

Comment (jayasanka-sack) [reply admin-portal/src/pages/needUpload/needUpload.tsx:37]: If the response comes with a response, then you should create an interface and provide it using generics:

ex: 
error instanceof AxiosError<X>

Comment (jayasanka-sack) [reply admin-portal/src/pages/needUpload/needUpload.tsx:37]: it should be 

```
catch (e){
if (!Object.prototype.hasOwnProperty.call(error, 'isAxiosError')) {
     const error = error as AxiosError<Y>;
}
}
```

`Y` should be the type of response payload.

File: admin-portal/src/pages/needUpload/needUpload.tsx [hunk 6]
```diff
@@ -77,13 +73,18 @@ export function NeedUpload() {
...
+              onChange={handleChange}
+              value={fileName}
+            />
             <button type="submit">Upload</button>
           </form>
         </div>
         <div className="error-list-div">
           {errorList.length > 0 &&
-            errorList.map((error: any) => <p>{error}</p>)}
+            errorList.map((error: string) => <p>{error}</p>)}
```
Comment (Gravewalker666) [inline admin-portal/src/pages/needUpload/needUpload.tsx:87]:
```suggestion
            errorList.map((error: string) => <p key={error}>{error}</p>)} 
```
or some key if possible

**Review by Gravewalker666 — CHANGES_REQUESTED** [review]

File: admin-portal/src/pages/needUpload/needUpload.tsx (hunk 1 above)
Comment (YaiyaSam) [reply admin-portal/src/pages/needUpload/needUpload.tsx]: Apparently the `Object.prototype` doesn't have the attribute `isAxiosError`. `AxiosError` has that property. So this actually returns false. Found out theres an easier way to do this. 

> catch (e) {
        if (axios.isAxiosError(e)) {
          const error = e as AxiosError<string>;
        }
   }

**Comment by YaiyaSam** [comment]
@jayasanka-sack sorry I forgot to leave a comment yesterday. I made the requested changes. Please review!

File: admin-portal/src/pages/needUpload/needUpload.tsx (hunk 6 above)
Comment (jayasanka-sack) [reply admin-portal/src/pages/needUpload/needUpload.tsx]: I don't think the `error` would be unique. We might need to create a unique key when we populate the array.

**Review by jayasanka-sack — CHANGES_REQUESTED** [review]
Thanks @YaiyaSam . Please check the following comment:
https://github.com/LSFLK/MedicinesforLK/pull/87#discussion_r917279912

**Comment by YaiyaSam** [comment]
> Thanks @YaiyaSam . Please check the following comment: [#87 (comment)](https://github.com/LSFLK/MedicinesforLK/pull/87#discussion_r917279912)

Fixed it!

File: admin-portal/src/pages/supplierQuotationUpload/supplierQuotationUpload.tsx [hunk 7]
```diff
@@ -79,13 +79,20 @@ export function SupplierQuotationUpload() {
...
+              value={fileName}
+            />
             <button type="submit">Upload</button>
           </form>
         </div>
         <div className="error-list-div">
-          {errorList.length > 0 &&
-            errorList.map((error: any) => <p>{error}</p>)}
+          {responseData.length > 0 &&
+            responseData.map((error: string, index: number) => (
```
Comment (jayasanka-sack) [inline admin-portal/src/pages/supplierQuotationUpload/supplierQuotationUpload.tsx:93]: We should avoid using the index as a key. What we can do instead is generate a unique key before setting it to the state. 
Read more:
https://robinpokorny.medium.com/index-as-a-key-is-an-anti-pattern-e0349aece318

Or else we can just display the error payload as it is (with line breaks). 
ex: https://stackoverflow.com/questions/39325414/line-break-in-html-with-n

Comment (jayasanka-sack) [reply admin-portal/src/pages/supplierQuotationUpload/supplierQuotationUpload.tsx:93]: I prefer the second option.

Comment (YaiyaSam) [reply admin-portal/src/pages/supplierQuotationUpload/supplierQuotationUpload.tsx]: i split the string cuz the success message has vertical bars separating the lines. i'll ask anupama to use line breaks instead. i changed the code. pls review.

**Review by jayasanka-sack — APPROVED** [review]
Thanks @YaiyaSam !

**Pull request merged**